mysql/infoschema.go

```diff
@@ -267,21 +268,22 @@ func processColumns(conv *internal.Conv, cols *sql.Rows, constraints map[string]
 	return colDefs, colNames
 }
 
-// getConstraints returns a list of primary keys and by-column map of
-// other constraints.  Note: we need to preserve ordinal order of
-// columns in primary key constraints.
-func getConstraints(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]string, map[string][]string, error) {
+// getConstraints returns a list of primary keys, foreign keys and
+// by-column map of other constraints.  Note: we need to preserve
+// ordinal order of columns in primary key constraints.
+func getConstraints(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]string, map[string][]string, []schema.Fkey, error) {
```

Comment:
I'd be tempted to split out the foreign key functionality into a separate function -- say getForeignKeys.
The main reason is that all of the functionality for foreign keys is independent of the existing getConstraints functionality.

The only down side of this is that we'd have two functions:

getConstraints
getForeignKeys

and you could argue that foreign keys are also constraints.
I'd handle that by adding a comment in getConstraints something like:
// Note that foreign key constraints are handled elsewhere.

I think that shorter simpler functions outweigh this potential confusion.

Reply:
Got it. I will update it and follow same structure in postgres/infoschema.